File: PlacementCellBackend/DTOs/AlumniFeedBackOnCompanyDTO.cs
using PlacementCellBackend.Models.Enums;
using PlacementCellBackend.Models.InterviewRounds;

namespace PlacementCellBackend.DTOs;

public class AlumniFeedBackOnCompanyDTO
{
    public string companyName { get; set; } = string.Empty;

    public string AlumniProfile { get; set; } = string.Empty;

    public string jobProfile { get; set; } = string.Empty;

    public string CTC { get; set; } = string.Empty;

    public string JobLocation { get; set; } = string.Empty;

    public JobType JobType { get; set; } = JobType.Internship;

    public WorkMode WorkMode { get; set; } = WorkMode.Remote;

    public CodingRound? CodingRoundInfo { get; set; }

    public TechnicalRound? TechnicalRoundInfo { get; set; }

    public HRRound? HRRoundInfo { get; set; }

    public Resources? ResourcesInfo { get; set; }
}

public class AlumniFeedBackOnCompanyCreateDTO
{
    public string companyid { get; set; } = string.Empty;
    public string alumniid { get; set; } = string.Empty;
    public string jobProfile { get; set; } = string.Empty;
    public string CTC { get; set; } = string.Empty;
    public string JobLocation { get; set; } = string.Empty;
    public JobType JobType { get; set; } = JobType.Internship;
    public WorkMode WorkMode { get; set; } = WorkMode.Remote;
    public CodingRound? CodingRoundInfo { get; set; }
    public TechnicalRound? TechnicalRoundInfo { get; set; }
    public HRRound? HRRoundInfo { get; set; }
    public Resources? ResourcesInfo { get; set; }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/dtos/AlumniFeedBackOnCompanyDTO.java
package com.example.placementicsbackend.dtos;

import com.example.placementicsbackend.models.Enums.JobType;
import com.example.placementicsbackend.models.Enums.WorkMode;
import com.example.placementicsbackend.models.InterviewRounds.CodingRound;
import com.example.placementicsbackend.models.InterviewRounds.HRRound;
import com.example.placementicsbackend.models.InterviewRounds.Resources;
import com.example.placementicsbackend.models.InterviewRounds.TechnicalRound;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlumniFeedbackOnCompanyDto {

    private String companyName = "";
    private String alumniProfile = "";
    private String jobProfile = "";
    private String ctc = "";
    private String jobLocation = "";
    private JobType jobType = JobType.Internship;
    private WorkMode workMode = WorkMode.Remote;
    private CodingRound codingRoundInfo;
    private TechnicalRound technicalRoundInfo;
    private HRRound hrRoundInfo;
    private Resources resourcesInfo;
}
